File: frontend/src/types/finance.ts
export interface Expense {
  id: string;
  workspace_id: string;
  project_id?: string | null;
  title: string;
  amount: number;
  category: string;
  expense_date: string;
  notes?: string | null;
  created_at: string;
}

export interface Budget {
  id: string;
  workspace_id: string;
  category: string;
  amount: number;
  period: string;
  start_date: string;
}

export interface FinancialAccount {
  id: string;
  workspace_id: string;
  account_name: string;
  current_cash_balance: number;
  currency: string;
  last_updated_at: string;
}

export interface BurnRateData {
  monthly_burn_rate: number;
  total_historical_spend: number;
  period_days: number;
  calculation_method: string;
}

export interface RunwayData {
  has_cash_data: boolean;
  available_cash: number;
  monthly_burn_rate: number;
  runway_months: number | null;
  currency?: string;
  message: string;
}

export interface BudgetComparison {
  category: string;
  budget: number;
  actual: number;
  variance: number;
  percent_used: number;
  is_exceeded: boolean;
}

File: backend/app/schemas/finance.py
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field


class ExpenseCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    title: str = Field(..., min_length=1, max_length=255)
    amount: float = Field(..., gt=0)
    category: str = "OPERATIONS"
    project_id: Optional[str] = None
    expense_date: Optional[datetime] = None
    notes: Optional[str] = None


class ExpenseResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True, extra="ignore")

    id: str
    workspace_id: str
    project_id: Optional[str] = None
    title: str
    amount: float
    category: str
    expense_date: datetime
    notes: Optional[str] = None
    created_at: datetime


class BudgetCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    category: str
    amount: float = Field(..., gt=0)
    period: str = "MONTHLY"


class BudgetResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True, extra="ignore")

    id: str
    workspace_id: str
    category: str
    amount: float
    period: str
    start_date: datetime


class CashBalanceUpdate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    balance: float = Field(..., ge=0)
    currency: str = "USD"


class FinancialAccountResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    workspace_id: str
    account_name: str
    current_cash_balance: float
    currency: str
    last_updated_at: datetime
